src/main/scala/com/microsoft/ml/spark/vw/featurizer/StringSplitFeaturizer.scala

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark.featurizer
+
+import com.microsoft.ml.spark.VowpalWabbitMurmurWithPrefix
+import org.apache.spark.sql.Row
+import org.vowpalwabbit.spark.VowpalWabbitMurmur
+
+import scala.collection.mutable.ArrayBuilder
+
+/**
+  * Featurize strings by splitting into native VW structure. (hash(s(0)):value, hash(s(1)):value, ...)
+  * @param fieldIdx input field index.
+  * @param columnName used as feature name prefix.
+  * @param namespaceHash pre-hashed namespace.
+  * @param mask bit mask applied to final hash.
+  */
+class StringSplitFeaturizer(override val fieldIdx: Int, val columnName: String, val namespaceHash: Int, val mask: Int)
+  extends Featurizer(fieldIdx) {
+
+  /**
+    * (?U) makes \w unicode aware
+    * https://stackoverflow.com/questions/4304928/unicode-equivalents-for-w-and-b-in-java-regular-expressions
+    * we could follow
+    * https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.CountVectorizer.html
+    * but that strips single character words...
+    */
+  val nonWhiteSpaces = "(?U)\\w+".r
```

Comment:
in the future might be nice to allow the user to specify this

Reply:
eg make it a user param

Reply:
added TODO